examples/seq2seq/run_sum.py

```diff
@@ -265,6 +250,13 @@ def main():
     else:
         model_args, data_args, training_args = parser.parse_args_into_dataclasses()
 
+    # @patrickvonplaten|@stas00 picked this up from
+    #   https://github.com/huggingface/transformers/issues/10164#issuecomment-779293173
+    #   https://github.com/huggingface/transformers/pull/10611#discussion_r590595798
+    # but unclear for me if I should restrict like this or just `.startswith("t5")`?
+    if data_args.source_prefix is None and model_args.model_name_or_path in ["t5-small", "t5-base", "t5-large", "t5-3b", "t5-11b"]:
+        data_args.source_prefix = "summarize: "
```

Comment:
could you please check if the same is needed for mt5?

Perhaps a cleaner solution here would be to refer to the config object?

e.g. get the config object and then check `if config.architecture in [ "MT5ForConditionalGeneration", "T5WithLMHeadModel"]` I don't know if it's exhaustive.

I'm just looking at:

https://huggingface.co/t5-small/blob/main/config.json#L3
https://huggingface.co/google/mt5-small/blob/main/config.json#L3

Reply:
Regarding the T5 prefixes:

We had discussed this in a previous PR and I think @patrickvonplaten and I agreed that it's best to just leave it to the user to specify the correct prefix and avoid having such if/else logic. And add a note about that in the readme, so the user will know when and what prefix to pass. I'm a bit unsure why it's being added back.

Reply:
Yes that perspective is relevant to the tradeoff between simpler scripts vs ready-to-use scripts, as I outlined here: https://github.com/huggingface/transformers/pull/10611#discussion_r591169232

Reply:
The problem is that it leads to:
1. To multiple points of language pair definition which leads to error-prone code. The language pair should be specified once
2. The software shouldn't subtly misbehave if some required flag is passed or not. At the very least the example script should require `--source_prefix` cl arg if the model is of type `*t5*` - it must not be optional.

Combining these 2 together is what @theo-m  implemented.

Reply:
I'm voting against this. 3 reasons from my side:

- This is not in line with "simple notebooks" that are easy to tweak. I think we somewhat decided that notebooks should not follow the one-size-fits-it-all approach.
- We add those lines only for 5 checkpoints, while we could very well just let the user define the prefix. A user that knows how T5 was trained can insert the prefix. Also, for a user that doesn't know T5 this happens automatically which could mean that she/he might not be aware afterward that a prefix has to be added to make summarization work.
- People that fine-tune a model like `t5-large` don't necessarily need the summarization prefix to work. `t5-large` can very well be fine-tuned without a prefix (not the strongest argument).

Reply:
This is also not scalable when we have other "multi-task" models that work for summarization. There is a very low chance that those need the same prefix.

Reply:
I think I would be fine with a warning if you guys think it's really important